src/irmin-pack/unix/file_manager.ml

```diff
@@ -309,15 +309,16 @@ struct
     in
     finish_constructing_rw config control ~make_dict ~make_suffix ~make_index
 
+  let decode_int63 buf = Int63.decode ~off:0 buf
```

Comment:
I don't understand why this is necessary 🤔

Reply:
if it helps, see https://github.com/mirage/optint/blob/master/src/int63_native.ml#L74 vs https://github.com/mirage/optint/blob/master/src/int63_native.ml#L34 (but I don't understand either)

Reply:
Isn't it just that "of_string_opt" works with strings like "1234" but won't work with strings/bufs that have been written using Int63.encode/decode?

Reply:
Ah right. `of_string` is the inverse of `pp`